server/channels/app/scheduled_post.go

```diff
@@ -91,3 +91,26 @@ func (a *App) UpdateScheduledPost(rctx request.CTX, userId string, scheduledPost
 
 	return scheduledPost, nil
 }
+
+func (a *App) DeleteScheduledPost(rctx request.CTX, userId, scheduledPostId, connectionId string) (*model.ScheduledPost, *model.AppError) {
+	scheduledPost, err := a.Srv().Store().ScheduledPost().Get(scheduledPostId)
+	if err != nil {
+		return nil, model.NewAppError("app.DeleteScheduledPost", "app.delete_scheduled_post.get_scheduled_post.error", map[string]any{"user_id": userId, "scheduled_post_id": scheduledPostId}, "", http.StatusInternalServerError)
+	}
+
+	if scheduledPost == nil {
+		return nil, model.NewAppError("app.DeleteScheduledPost", "app.delete_scheduled_post.existing_scheduled_post.not_exist", map[string]any{"user_id": userId, "scheduled_post_id": scheduledPostId}, "", http.StatusNotFound)
+	}
+
+	if scheduledPost.UserId != userId {
+		return nil, model.NewAppError("app.DeleteScheduledPost", "app.delete_scheduled_post.delete_permission.error", map[string]any{"user_id": userId, "scheduled_post_id": scheduledPostId}, "", http.StatusForbidden)
+	}
```

Comment:
I think I already made this question (or similar) many times, but please bear with me.

Always forbidden? I am thinking about mmctl or admin operations, but 0/5.

Reply:
`403 Forbidden` should be correct here - the user is logged in but they're trying to delete a scheduled post that isn't theirs.

Reply:
@larkox for mmctl or admin operations, if, and when we add it, we'll handle those in the `local mode` like we do in other places. Allowing it right now could cause unnecessary security issues.